src/actions/vis-state-actions.js

```diff
@@ -508,6 +508,76 @@ export function setFeatures(features) {
   }
 }
 
+/**
+ * It will apply the provide feature as filter to the given layer.
+ * If the given feature is already applied as filter to the layer it  will remove it from  the filter list
+ * @memberof visStateActions
+ * @param {Object} featureId feature to use as filter base for layer
+ * @param {Object} layer
+ * @return {{feature: *, type: ActionTypes.TOGGLE_POLYGON_FILTER, layer: *}}
+ */
+export function togglePolygonFilter(layer, featureId) {
```

Comment:
the naming of this action is a bit blurry. it sounds like this is to hide and show polygon filter. If layer is passed along can it be `setPolygonFilterLayer` ?